pages/nfts/new.tsx

```diff
@@ -0,0 +1,323 @@
+  share: number;
+}
+
+const StyledLayout = styled(Layout)`
+  width: 100%;
+  overflow: hidden;
+`;
+
+export interface UploadedFilePin {
+  name: string;
+  uri: string;
+  type: string;
+}
+
+export type FormValues = { [key: string]: NFTFormValue };
+
+export interface NFTFormValue {
+  name: string;
+  imageName: string;
+  description: string;
+  collection: string;
+  attributes: Array<NFTAttribute>;
+  seller_fee_basis_points: number;
+  properties: { creators: Array<Creator>; maxSupply: number };
+}
+
+export type FileOrString = UploadedFilePin | string;
+
+export type NFTAttribute = {
+  trait_type: string;
+  value: string | number;
+};
+
+export enum MintStatus {
+  FAILED,
+  SUCCESS,
+}
+
+export interface NFTValue {
+  name: string;
+  description: string;
+  attributes?: NFTAttribute[];
+  collection?: string;
+  seller_fee_basis_points: number;
+  mintStatus?: MintStatus;
+
+  properties: {
+    files?: FileOrString[];
+    maxSupply: number;
+    creators?: Creator[];
+  };
+}
+
+export type MintDispatch = (action: MintAction) => void;
+
+interface State {
+  images: Array<File>;
+  uploadedFiles: Array<UploadedFilePin>;
+  formValues: NFTFormValue[] | null;
+  nftValues: NFTValue[];
+}
+
+export interface MintAction {
+  type:
+    | 'SET_IMAGES'
+    | 'DELETE_IMAGE'
+    | 'ADD_IMAGE'
+    | 'UPLOAD_FILES'
+    | 'SET_FORM_VALUES'
+    | 'SET_NFT_VALUES';
+  payload: File[] | File | String | Array<UploadedFilePin> | NFTFormValue[] | NFTValue[];
+}
+
+const initialState: State = {
+  images: [],
+  uploadedFiles: [],
+  formValues: null,
+  nftValues: [],
+};
+
+function reducer(state: State, action: MintAction) {
+  switch (action.type) {
+    case 'SET_IMAGES':
+      return { ...state, images: action.payload as File[] };
+    case 'DELETE_IMAGE':
+      return {
+        ...state,
+        images: state.images.filter((i) => i.name !== (action.payload as String)),
+      };
+    case 'ADD_IMAGE':
+      return state.images.length < 10
+        ? { ...state, images: [...state.images, action.payload as File] }
+        : state; // since the dispactch is async(?) it did not reliably limit number of files from the Upload component.
+    case 'UPLOAD_FILES':
+      return { ...state, uploadedFiles: action.payload as Array<UploadedFilePin> };
+    case 'SET_FORM_VALUES':
+      return { ...state, formValues: action.payload as NFTFormValue[] };
+    case 'SET_NFT_VALUES':
+      return { ...state, nftValues: action.payload as NFTValue[] };
+    default:
+      throw new Error('No valid action for state');
+  }
+}
+
+const connection = new Connection(process.env.NEXT_PUBLIC_SOLANA_ENDPOINT as string);
+
+export default function BulkUploadWizard() {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const [form] = useForm();
+  const { connect, solana, wallet, storefront } = useContext(WalletContext);
+  const { images, formValues } = state;
+
+  const [doEachRoyaltyInd, setDoEachRoyaltyInd] = React.useState(false);
```

Comment:
`useState` should be imported from `'react'` at the top of the file for conformity.
```suggestion
  const [doEachRoyaltyInd, setDoEachRoyaltyInd] = useState(false);
```